.grit/patterns/mux_v8.md

```diff
@@ -0,0 +1,278 @@
+
+                    return fetch(url, opts)
+                }`
+            }
+        },
+    }
+}
+
+pattern change_constructors() {
+    `$destruct = new Mux($params)` where {
+        $destruct <: contains `{ $props }` => `mux`,
+        $props <: maybe some change_destructured_property_call(),
+        $params <: or {
+            [$tokenId, $tokenSecret, $config] where {
+                $config <: convert_config() as $parsed_config where {
+                    $params => `{
+                        tokenId: $tokenId,
+                        tokenSecret: $tokenSecret,
+                        $parsed_config
+                    }`
+                }
+            },
+            [$tokenId, $tokenSecret] => `{
+                tokenId: $tokenId,
+                tokenSecret: $tokenSecret,
+            }`,
+            convert_config() as $config => $config,
+            . => .,
+        },
+    },
+}
+
+pattern as_lower_camel_case($formatted) {
+    r"([A-Z])([a-zA-Z]*)"($first_char, $rest) where {
+        $first_char = lowercase(string = $first_char),
+        $formatted = join(list = [$first_char, $rest], separator = ""),
+    }
+}
+
+pattern change_destructured_property_call() {
+    $prop where {
+        $program <: contains bubble($prop) or {
+            `$prop.$field.$action($arg)` => `mux.$prop.$field.$action($arg)`,
+            `$prop.$field.$action($arg, $opt)` => `mux.$prop.$field.$action($arg, $opt)`,
+        } where {
+            $formatted = ``,
+            $prop <: as_lower_camel_case($formatted) where {
+                $prop => `$formatted`
+            },
+            $field <: as_lower_camel_case($formatted) where {
+                $field => `$formatted`
+            },
+            $prop <: maybe `Video` where or {
+                $field <: `Assets`,
+                $field <: `DeliveryUsage`,
+                $field <: `LiveStreams`,
+                $field <: `PlaybackIDs`,
+                $field <: `PlaybackRestrictions`,
+                $field <: `Spaces`,
+                $field <: `TranscriptionVocabularies`,
+                $field <: `Uploads`,
+            } where {
+                $action <: `get` where {
+                    $action => `retrieve`,
+                }
+            },
+            $prop <: maybe `Video` where {
+                $field <: or {
+                    `Assets`,
+                    `LiveStreams`,
+                    `Uploads`
+                } where {
+                    $arg <: $data where {
+                        $data <: maybe contains `new_asset_settings: $new_asset_settings` where {
+                            $new_asset_settings <: contains `playback_policy: $playback_policy` where {
+                                $playback_policy <: string(fragment=$_) => `[$playback_policy]`,
+                            }
+                        },
+                        $data <: maybe contains `playback_policy: $playback_policy` where {
```

Comment:
Isn't this redundant with the above change?

Reply:
This is still required because the first
```
$data <: maybe contains `new_asset_settings: $new_asset_settings`
```
and the second
```
$data <: maybe contains `playback_policy: $playback_policy`
```
both match on `$data` but with different key values.